### PR TITLE
feat: translate web-components from header to inputs

### PR DESCRIPTION
**Pull request by flagrede**
## Description

translate web-components from header to inputs

Note for reviewers:
This [commit](https://github.com/regen-network/regen-web/pull/2468/commits/5115cd6aad0e02fbce9ccb0108a207c86257c9f0) is important as it extracts all the validation logic of forms that was in `web-components`.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-www/components/templates/Layout/Layout.tsx [hunk 1]
```diff
@@ -49,6 +53,8 @@ const Layout = ({ children }: Props) => {
               </Body>
             </>
           }
+          requiredMessage={REQUIRED_MESSAGE}
```
Comment (wgardiner) [inline web-www/components/templates/Layout/Layout.tsx:56]: needs translation with `_(...)`?

File: web-www/components/templates/shared/EmailSubmitSection/EmailSubmitSection.tsx [hunk 2]
```diff
@@ -70,6 +74,8 @@ const EmailSubmitSection = ({ altContent, sharedNewsletterData }: Props) => {
               </Body>
             </>
           }
+          requiredMessage={REQUIRED_MESSAGE}
```
Comment (wgardiner) [inline web-www/components/templates/shared/EmailSubmitSection/EmailSubmitSection.tsx:77]: needs translation with `_(...)`?

File: web-www/lib/constants/shared.constants.ts [hunk 3]
```diff
@@ -6,3 +6,5 @@ export const ARTICLE_CARD_BTN_TEXT_MAPPING: Record<ArticleType, string> = {
   article: 'read article',
   podcast: 'listen to podcast',
 };
+export const REQUIRED_MESSAGE = 'This field is required';
+export const INVALID_EMAIL_MESSAGE = 'Please enter a valid email address';
```
Comment (wgardiner) [inline web-www/lib/constants/shared.constants.ts:10]: needs translation

File: web-www/pages/contact.tsx [hunk 4]
```diff
@@ -81,18 +81,18 @@ export default function ContactPage({
...
-                    errors.name = requiredMessage;
+                    errors.name = REQUIRED_MESSAGE;
                   }
                   if (!values.requestType) {
-                    errors.requestType = requiredMessage;
+                    errors.requestType = REQUIRED_MESSAGE;
                   }
                   if (!values.message) {
-                    errors.message = requiredMessage;
+                    errors.message = REQUIRED_MESSAGE;
```
Comment (wgardiner) [inline web-www/pages/contact.tsx:95]: needs translation

**Review by wgardiner — COMMENTED** [review]
Looks good overall, just one typo in a variable name to fix, and a few missing translations.

File: web-components/src/components/form/CreditRetireForm.tsx [hunk 5]
```diff
@@ -224,7 +229,11 @@ export const BottomCreditRetireFields: React.FC<
               />
             }
           >
-            <LocationCountryField exclude name={`${arrayPrefix}country`} />
+            <LocationCountryField
+              exclude
+              name={`${arrayPrefix}country`}
```
[the remark this answers is not in the document]
Comment (flagrede) [reply web-components/src/components/form/CreditRetireForm.tsx:234]: This is the name of the input, it's not meant to be translated.

File: web-www/components/templates/Layout/Layout.tsx (hunk 1 above)
Comment (flagrede) [reply web-www/components/templates/Layout/Layout.tsx:56]: It's in `web-www`; as discussed, it's out of scope for now.

File: web-www/components/templates/shared/EmailSubmitSection/EmailSubmitSection.tsx (hunk 2 above)
Comment (flagrede) [reply web-www/components/templates/shared/EmailSubmitSection/EmailSubmitSection.tsx:77]: https://github.com/regen-network/regen-web/pull/2468#discussion_r1761096649

File: web-www/lib/constants/shared.constants.ts (hunk 3 above)
Comment (flagrede) [reply web-www/lib/constants/shared.constants.ts:10]: https://github.com/regen-network/regen-web/pull/2468#discussion_r1761096649

File: web-www/pages/contact.tsx (hunk 4 above)
Comment (flagrede) [reply web-www/pages/contact.tsx:95]: https://github.com/regen-network/regen-web/pull/2468#discussion_r1761096649

**Review by wgardiner — APPROVED** [review]

**Pull request merged**